cmd/skywire-cli/commands/vpn/vvpn.go

```diff
@@ -0,0 +1,191 @@
+package clivpn
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/toqueteos/webbrowser"
+
+	clirpc "github.com/skycoin/skywire/cmd/skywire-cli/commands/rpc"
+	clivisor "github.com/skycoin/skywire/cmd/skywire-cli/commands/visor"
+	"github.com/skycoin/skywire/cmd/skywire-cli/internal"
+	"github.com/skycoin/skywire/pkg/servicedisc"
+	skyenv "github.com/skycoin/skywire/pkg/skyenv"
+	"github.com/skycoin/skywire/pkg/transport/network"
+	"github.com/skycoin/skywire/pkg/visor"
+	"github.com/skycoin/skywire/pkg/visor/visorconfig"
+)
+
+var timeout time.Duration
+
+func init() {
+	RootCmd.AddCommand(
+		vpnListCmd,
+		vpnUICmd,
+		vpnURLCmd,
+		vpnStartCmd,
+		vpnStopCmd,
+	)
+	vpnListCmd.Flags().StringVarP(&ver, "ver", "v", "1.0.1", "filter results by version")
+	vpnListCmd.Flags().StringVarP(&country, "country", "c", "", "filter results by country")
+	vpnListCmd.Flags().BoolVarP(&stats, "stats", "s", false, "return only a count of the resuts")
+	vpnListCmd.Flags().BoolVarP(&systray, "systray", "y", false, "format results for systray")
+}
+
+var vpnUICmd = &cobra.Command{
+	Use:   "ui",
+	Short: "Open VPN UI in default browser",
+	Run: func(_ *cobra.Command, _ []string) {
+		var url string
+		if pkg {
+			path = visorconfig.Pkgpath
+		}
+		if path != "" {
+			conf, err := visorconfig.ReadFile(path)
+			if err != nil {
+				log.Fatal("Failed to read in config:", err)
+			}
+			url = fmt.Sprintf("http://127.0.0.1:8000/#/vpn/%s/", conf.PK.Hex())
+		} else {
+			client := clirpc.RPCClient()
+			overview, err := client.Overview()
+			if err != nil {
+				log.Fatal("Failed to connect; is skywire running?\n", err)
+			}
+			url = fmt.Sprintf("http://127.0.0.1:8000/#/vpn/%s/", overview.PubKey.Hex())
+		}
+		if err := webbrowser.Open(url); err != nil {
+			log.Fatal("Failed to open VPN UI in browser:", err)
+		}
+	},
+}
+
+var vpnURLCmd = &cobra.Command{
+	Use:   "url",
+	Short: "Show VPN UI URL",
+	Run: func(_ *cobra.Command, _ []string) {
+		var url string
+		if pkg {
+			path = visorconfig.Pkgpath
+		}
+		if path != "" {
+			conf, err := visorconfig.ReadFile(path)
+			if err != nil {
+				log.Fatal("Failed to read in config:", err)
+			}
+			url = fmt.Sprintf("http://127.0.0.1:8000/#/vpn/%s/", conf.PK.Hex())
+		} else {
+			client := clirpc.RPCClient()
+			overview, err := client.Overview()
+			if err != nil {
+				logger.Fatal("Failed to connect; is skywire running?\n", err)
+			}
+			url = fmt.Sprintf("http://127.0.0.1:8000/#/vpn/%s/", overview.PubKey.Hex())
+		}
+		fmt.Println(url)
+	},
+}
+
+var vpnListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List public VPN servers",
+	Run: func(_ *cobra.Command, _ []string) {
+		client := clirpc.RPCClient()
+		servers, err := client.VPNServers()
+		if err != nil {
+			logger.Fatal("Failed to connect; is skywire running?\n", err)
+		}
+		var a []servicedisc.Service
```

Comment:
What are we trying to do here? I think the inconsistent application of `v` or not `v` should be validated in the service discovery instead. I dont seem to understand why we are checking for unknown or empty versions and then appending them to the slice.